l10n_fr_ecotaxe_sale/tests/test_sale_ecotaxe.py

```diff
@@ -0,0 +1,124 @@
+        )
+        # I assign a product with fixed ecotaxte to sale line
+        self.sale_line1 = self.sale.order_line[0]
+        self.product_a.manual_fixed_ecotaxe = 1.5
+        self.sale_line1._compute_ecotaxe()
+        self.assertEqual(self.product_a.ecotaxe_amount, 1.5)
+        self.sale_line1.product_uom_qty = 4
+        self.sale_line1.product_uom_change()
+        self.assertEqual(self.sale_line1.unit_ecotaxe_amount, 1.5)
+        self.assertEqual(self.sale_line1.subtotal_ecotaxe, 6.0)
+        self.assertEqual(self.sale.amount_total, 400.0)
+        self.assertEqual(self.sale.amount_ecotaxe, 6.0)
+
+    def test_02_classification_weight_based_ecotaxe(self):
+        """Tests multiple lines with weight based ecotaxes"""
+        # in order to test the correct assignment of fixed ecotaxe
+        # I create a customer sale.
+        partner12 = self.env.ref("base.res_partner_12")
+        self.sale = self.create_sale_partner(
+            sale_amount=100.00, partner_id=partner12, product_id=self.product_b
+        )
+        # I assign a product with fixed ecotaxte to sale line
+        self.sale_line1 = self.sale.order_line[0]
+        self.sale_line1._compute_ecotaxe()
+        self.assertEqual(self.product_b.ecotaxe_amount, 16)
+        self.sale_line1.product_uom_qty = 3
+        self.assertEqual(self.sale_line1.unit_ecotaxe_amount, 16)
+        self.assertEqual(self.sale_line1.subtotal_ecotaxe, 48)
+        self.assertEqual(self.sale.amount_total, 300.0)
+        self.assertEqual(self.sale.amount_ecotaxe, 48)
+
+    def test_03_classification_ecotaxe(self):
+        """Tests multiple lines with mixed ecotaxes"""
+        # in order to test the correct assignment of fixed ecotaxe
+        # I create a customer sale.
+        partner12 = self.env.ref("base.res_partner_12")
+        self.sale = self.create_sale_partner(
+            sale_amount=100.00, partner_id=partner12, product_id=self.product_a
+        )
+        # add line to SO
+        self.env["sale.order.line"].create(
+            {
+                "product_id": self.product_b.id,
+                "product_uom_qty": 2,
+                "price_unit": 100,
+                "order_id": self.sale.id,
+            },
+        )
+        # I assign a product with fixed ecotaxte to sale line
+        self.sale_line1 = self.sale.order_line[0]
+        self.sale_line2 = self.sale.order_line[1]
+        self.sale_line1._compute_ecotaxe()
+        self.assertEqual(self.product_a.ecotaxe_amount, 2.4)
+        self.sale_line1.product_uom_qty = 3
+        self.sale_line1.product_uom_change()
+        self.assertEqual(self.sale_line1.unit_ecotaxe_amount, 2.4)
+        self.assertAlmostEqual(self.sale_line1.subtotal_ecotaxe, 7.2)
+        self.assertEqual(self.sale_line2.unit_ecotaxe_amount, 16)
+        self.assertEqual(self.sale_line2.subtotal_ecotaxe, 32)
+        self.assertEqual(self.sale.amount_total, 4300.0)
```

Comment:
@ivantodorovich  i am bit worried about assertions here, in this tc we have 2 sale order lines for product_a and product_b with qty 1 and 3, on each line product price is 100.00, don't get the point where total 3k - 4k comes from? Is it because of the weight computation for ecotax?